### ISSUE TITLE
Behaviour of the loss function with sample_weights (keras)

### ISSUE THREAD

**Issue by Emilien-P** [tensorflow/tensorflow#34305]
## URL(s) with the issue:
https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit

## Description of issue (what needs changing):
I think we need a description of how the loss is computed when (temporal) sample_weights are applied, i.e. how is the loss aggregated across samples/time-steps. As addressed in #25178, the behaviour has even changed over time in the external version of Keras from ignoring zero-values to counting them making the situation confusing. 

Even if I agree the current implementation is the right one, I think it is far from intuitive, especially when using zero-weighted zero-padding which decreases the loss value and the effective learning rate.

I am not very familiar with the structure to be followed in the build-in TF.Keras documentation, but this could be also specified elsewhere than in the `fit` method doc which is already pretty dense.

Sorry if this is specified elsewhere in the doc but I can't find it so if it exists, it should maybe be referenced in the `fit` doc.
Thanks a lot,
Emilien

**Comment by dubesar** [comment]
Hey!!
I confirmed the #25178 and still, it works inconsistently. But as the solution mentioned there is to use tf-nightly. If that too doesn't work please ping again.

**Comment by Emilien-P** [comment]
Hi,
Thanks for having a look :)
I think the main issue now is about documenting the issue so that we are aware of whatever loss aggregation function is chosen.

**Comment by lamberta** [comment]
Sounds like a good suggestion!
Can you add it to the docstring here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/engine/training.py#L596
We now support DocTest snippets in the API reference so that might be a good place to start: https://www.tensorflow.org/community/contribute/docs_ref

**Comment by Emilien-P** [comment]
@lamberta Thanks for the references, I will make a doc edition proposal :)

**Comment by Saduf2019** [comment]
@Emilien-P 
Is this still an issue.
